src/main/java/io/connect/scylladb/RecordConverter.java

```diff
@@ -55,30 +58,56 @@
 
     protected abstract void setNullField(T result, String name);
 
-    public T convert(Object value) {
-        Preconditions.checkNotNull(value, "value cannot be null.");
+    public T convert(SinkRecord record, TopicConfigs topicConfigs, String operationType) {
+        Object recordObject = "delete".equals(operationType) ? record.key() : record.value();
```

Comment:
"delete" should be a static variable defined ideally in some common class
(and used below)